request.js

```diff
@@ -921,7 +921,7 @@ Request.prototype.onRequestResponse = function (response) {
     })
 
     var responseContent
-    if (self.gzip) {
+    if (self.gzip && self.method !== 'HEAD') {
```

Comment:
You can make assumption in your own layer, that when you receive that error under certain circumstances it may mean something, but as far as request is concerned the error should be forwarded.
I'm open for suggestion though.